tests/test_client.py

```diff
@@ -33,7 +33,10 @@ def get_session():
 
     @app.route('/sessions', methods=['POST'])
     def create_session():
-        assert request.get_json() == {'kind': 'pyspark'}
+        data = request.get_json()
+        assert 'kind' in data and data['kind'] == 'pyspark'
+        if 'conf' in data:
+            assert isinstance(data['conf'], dict)
```

Comment:
To simplify the test logic a little here,, we could just have a single test for `create_session` which always passes a `spark_conf` and additionally verifies its contents.

There could be a module-level variable like:

```python
TEST_SPARK_CONF = {
    "spark.master": "yarn",
    "spark.submit.deployMode": "client"
}
```

which is used both in the main test:

```python
def test_create_session(mocker, server): 
    mocker.patch.object(Session, 'from_json')
    client = LivyClient(server)
    session = client.create_session(
        SessionKind.PYSPARK,
        spark_conf=TEST_SPARK_CONF
    )
```

and in the assertion in the test server:

```python
assert request.get_json() == {
    'kind': 'pyspark',
    'conf': TEST_SPARK_CONF
}
```

What do you think about something like this?